src/utils/fundingCycle.test.ts

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, test } from 'vitest'
+import { getNextCycleWeight, getPrevCycleWeight } from './fundingCycle';  
+import { MAX_DISCOUNT_RATE } from '../constants'; 
+
+describe("Cycle Weight utilities", () => {
+
+  test.each`
+    weight      | discountRate  | expected
+    ${100n}     | ${20n}        | ${(100n * (MAX_DISCOUNT_RATE - 20n)) / MAX_DISCOUNT_RATE}
+    ${200n}     | ${30n}        | ${(200n * (MAX_DISCOUNT_RATE - 30n)) / MAX_DISCOUNT_RATE}
+    ${300n}     | ${0n}         | ${(300n * (MAX_DISCOUNT_RATE - 0n)) / MAX_DISCOUNT_RATE}
+  `("computes next cycle weight correctly", ({ weight, discountRate, expected }) => {
+    expect(getNextCycleWeight({ weight, discountRate })).toEqual(expected);
+  });
+
+  test.each`
+    weight      | discountRate  | expected
+    ${80n}      | ${20n}        | ${(80n * MAX_DISCOUNT_RATE) / (MAX_DISCOUNT_RATE - 20n)}
```

Comment:
@johnnyd-eth in this case I think making this a literal value (the result of the expression) rather than the expression might be a bit safer. That way we're checking an explicit value, not the result of some other formula that's defined in a test file. WDYT?

Reply:
Good call. I targeted the wrong branch on this PR, so this comment is actually relevant to another PR in this stack: https://github.com/peeldao/juice-hooks/pull/8#issuecomment-1788678452. Have fixed on that one 🙏